linera-views/src/collection_view.rs

```diff
@@ -351,6 +351,103 @@ where
     }
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+impl<'a, C, W> ByteCollectionView<C, W>
+where
+    C: Context + Send + Clone + 'static,
+    ViewError: From<C::Error>,
+    W: View<C> + Send + 'static,
+{
+    /// Pre load the list of entries into the database in order to have direct access to the views.
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::{create_test_context, MemoryContext};
+    /// # use linera_views::collection_view::ByteCollectionView;
+    /// # use linera_views::register_view::RegisterView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_context();
+    ///   let mut coll : ByteCollectionView<_, RegisterView<_,String>> = ByteCollectionView::load(context).await.unwrap();
+    ///   coll.pre_load_multi_entry(vec![vec![0,1],vec![2,3]]).await.unwrap();
+    ///   let view = coll.load_entry_mut(vec![0,1]).await.unwrap();
+    ///   let value = view.get_mut();
+    ///   assert_eq!(*value, String::default());
+    /// # })
+    /// ```
+    pub async fn pre_load_multi_entry(
+        &'a mut self,
+        vector_short_key: Vec<Vec<u8>>,
+    ) -> Result<(), ViewError> {
```

Comment:
As discussed offline, it would be a safer API if we could merge `pre_load` and `direct_load` and return a vector of views here. Bonus point for returning a vector of futures on views (probably using `BoxFuture`).

Reply:
But now I understand that this requires ReentrantCollectionView..

Reply:
So, it is a question of whether this API makes any sense:
* It is relatively advanced and leaks the abstraction of views outside.
* It might be faster than the `ReentrantCollectionView` because of the internal structure (which would need to be checked). But that would be another story.
* It would confuse users.

So, unless there is a confirmation, I think the best is to cancel this PR.